addons/website/models/website_page.py

```diff
@@ -196,6 +196,8 @@ def unlink(self):
             if not pages_linked_to_iruiview and not page.view_id.inherit_children_ids:
                 # If there is no other pages linked to that ir_ui_view, we can delete the ir_ui_view
                 page.view_id.unlink()
+            # Manual cascade to let menu perform its own cleanup
+            page.menu_ids.unlink()
```

Comment:
Shouldn't we actually clear the menu cache instead if that's a cache issue?
It looks like we are trying to change code flow to avoid the cache issue instead of fixing the cache issue.
Or is it not enough to clear menu cache after Page unlink here?

Reply:
I don't have time to fully check this one today, I will check to ship this one tomorrow, but why isn't this raising the same error?
```python
self.env['website'].browse(1)._get_menu_ids()
# [4, 5, 6]
self.env['website.menu'].browse(6).page_id = 7
self.env['website.menu'].browse(6).page_id.unlink()
self.env['website'].browse(1)._get_menu_ids()
# [4, 5]
```

Reply:
I do not know what is page 7 in your environment. But if I drop the modified page through the shell, it keeps returning `[4, 5, 6]`.
If page 7 is a general view, without any specific one defined, it also unlinks the related `ir_ui_view` which will not be referenced anymore, and `website`'s `ir_ui_view`'s `unlink()` contains a clear cache which forces the recomputation of  `_get_menu_ids()`.

Reply:
I don't have the same results on my end.
Regardless if the page deletion will delete or not its view, I still have `_get_menu_ids()` returning the expected values (without the page's menu).

Isn't this minimal flow supposed to raise the same error?
![image](https://user-images.githubusercontent.com/30048408/143235706-63eec6af-faa7-40e8-86a4-6f89d248f7ff.png)

Setting the same view on 2 pages from 2 menus, is making the view not being removed (as you can see also in the logs, no remove on the view table).
Still, I have the correct value return.

I will investigate a bit.

Reply:
This scenario also unlinks an `ir.model.data` which also contains a `clear_caches()`.
In the bug scenario, there is no other page that shares the same view as the website-specific page (record 9 in the screenshot below), neither `ir.ui.view` nor `ir.model.data` gets unlinked, therefore the caches are not cleared.

![image](https://user-images.githubusercontent.com/71644421/143246334-197f33ed-2d3f-4543-b8a7-700d1597193d.png)

Reply:
Ok, that's what I was missing.

Now that everything has been figured, we can ship it. Let me know if you have some news from the ORM team about the cascade deletion and the cache clearing.

Reply:
![image](https://user-images.githubusercontent.com/71644421/143252527-2b87d266-8f6a-438e-b963-50c81f5222e8.png)

So I would say there is nothing very specific on the roadmap about this yet.